### ISSUE TITLE
String templates not colorized in VS 2013

### ISSUE THREAD

**Issue by mhegazy** [Microsoft/TypeScript#1698]
String templates should be colorizes like string literals.

**Comment by DanielRosenwasser** [comment]
Duplicate of #1477, see my comment here for an explanation of the technical difficulties associated with this: https://github.com/Microsoft/TypeScript/issues/1477#issuecomment-66907946 .

Though, we could enhance the lexical classifier.

**Comment by NoelAbrahams** [comment]
@DanielRosenwasser, colour would really help

![image](https://cloud.githubusercontent.com/assets/1106823/5799410/fcd36ab6-9fc9-11e4-8d13-4829d19a8a8a.png)

The template string version is more difficult to comprehend.

**Comment by DanielRosenwasser** [comment]
@NoelAbrahams I understand that. This is probably a lousy experience for most people, and as the one who implemented templates, I personally felt like I owed our users something a little better. This afternoon I whipped something up.

I have [a branch](https://github.com/Microsoft/TypeScript/tree/lexicalTemplateStrings) from which you can build the source code and [swap out your language service](https://github.com/Microsoft/TypeScript/issues/1110#issuecomment-62451204) in VS2013. The specific commit (e42ce9c3199c9449ac04e0ce09815d7f1e69db88) can be cherry picked onto `release-1.4`.

Basically, as long as your template strings' substitutions don't contain:
* Expressions with curly braces on a separate line.
* Other substitution template strings that span multiple lines.

this should fit your needs.

If it doesn't, well, you probably have other problems to worry about. :wink:

**Comment by NoelAbrahams** [comment]
@DanielRosenwasser, will try that out. Do you mean to say that in VS 2015 this should be coloured correctly? (A bug in NTVS is preventing us from using VS 2015 atm, but we should be migrating soon.)

Thanks!

**Comment by DanielRosenwasser** [comment]
You're very welcome! Let me know how this works out for you. Yes, 2015 should already work with template colorization.

That's unfortunate - does the Node Tools team know about the bug?

**Comment by NoelAbrahams** [comment]
Yes, there's a [discussion item](http://nodejstools.codeplex.com/discussions/571258).

**Comment by DanielRosenwasser** [comment]
A fix is now in `master`.
